defaults/defaults.go

```diff
@@ -162,3 +162,9 @@ var (
 	// CLIVersion is the software version of the Cilium CLI.
 	CLIVersion string
 )
+
+// NamespaceKey is the command context key for namespace.
+type NamespaceKey struct{}
+
+// K8sClientKey is the command context key for k8sClient.
+type K8sClientKey struct{}
```

Comment:
Like Casey already mentioned, I also wonder why these aren't typed. Why not:

```suggestion
// NamespaceKey is the command context key for namespace.
type NamespaceKey string

// K8sClientKey is the command context key for k8sClient.
type K8sClientKey string
```

But I might be missing something obvious.

Reply:
golangci-lint complains about using string as the context key 💭 

```
context-keys-type: should not use basic type string as key in context.WithValue (revive)
```

Reply:
Hm :thinking: interesting that the linter doesn't explain why we shouldn't use basic types :disappointed:

Reply:
Also interesting that the official Go doc example for `context.Context.WithValue` uses a type with underlying type `string`: https://pkg.go.dev/context#example-WithValue

Reply:
yeah https://go.dev/blog/context has some explanation, it's to avoid key collisions. i can do something similar 💭

Reply:
maybe something like this? https://gist.github.com/michi-covalent/a1ef2fa24a2b416670a5b5e97b436392

tobias how do you feel about adding a new `api` package in cilium-cli? i can't put this in `cli` package because it creates a circular dependency between https://github.com/cilium/cilium-cli/blob/pr/michi/majestic-ketchup/cli/cmd.go and https://github.com/cilium/cilium/blob/pr/michi/majestic-ketchup/pkg/cilium-cli/cmd/connectivity.go

Reply:
> maybe something like this? https://gist.github.com/michi-covalent/a1ef2fa24a2b416670a5b5e97b436392

LGTM. I guess https://gist.github.com/michi-covalent/a1ef2fa24a2b416670a5b5e97b436392#file-gistfile1-txt-L24 should usbe `k8sClientKey` instead of `namespaceKey` though.

> tobias how do you feel about adding a new `api` package in cilium-cli? i can't put this in `cli` package because it creates a circular dependency between https://github.com/cilium/cilium-cli/blob/pr/michi/majestic-ketchup/cli/cmd.go and https://github.com/cilium/cilium/blob/pr/michi/majestic-ketchup/pkg/cilium-cli/cmd/connectivity.go

Let's do it :rocket:

Reply:
updated the PR 🚀 🙏